enhancements/etcd/automated-cluster-backups.md

```diff
@@ -0,0 +1,137 @@
+---
+title: automated-cluster-backups
+
+authors:
+  - "@skolicha"
+  - "@dmace"
+  - "@hexfusion"
+reviewers:
+  - "@deads2k"
+  - "@hexfusion"
+approvers:
+  - TBD
+creation-date: 2020-06-04
+last-updated: 2020-06-04
+status: provisional
+see-also:
+  - "https://github.com/kubernetes/enhancements/blob/master/etcd/disaster-recovery-with-ceo.md"  
+---
+
+# automated-cluster-backups
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Current cluster backup procedure requires SSH and manual user commands from a node which are unsafe and error prone. 
+Furthermore, a periodic backup taken automatically will allow the user to restore from the most recent backup when 
+cluster gets into unexpected quorum loss or other unrecoverable data loss situations.
+
+## Motivation
+Customers are recommended to avoid sshing to nodes. On the other hand, requiring an ssh session on a host (master) to take 
+a backup goes against recommendations with SSH.
+
+The immediate need for automated backups is to provide a path to upgrade from 4.4 to 4.5, with the knowledge that if the
+upgrade fails for some reason, there is no easy path to rollback, as the etcd version 3.4.x (used in OCP 4.5) is 
+incompatible with the etcd version of 3.3.x (used in 4.4.x). Since restoring a backup is the safest way to rollback, it 
+is important to take automated periodical backups to protect the user from unexpected data loss in such scenarios.
+
+## Goals
+
+1.  Automatic periodic backups and automatic pruning
+1.  Make most recent backups readily available
+1.  Eliminate SSH requirement
+
+## Non-Goals
+
+*  Automated restoration is not considered for this enhancement.
+
+## Currently Supported Functionality 
+
+Currently the scripts available support the following functionality:
+### Cluster Backup 
+Takes a snapshot of cluster’s etcd data along with static-pod-resources at the time of the backup. 
+
+### Cluster Restore 
+Restores the etcd data from a backup snapshot. It also restores the static pod resources while deleting all the newer
+revisions.
+
+## Proposal
+
+### A new container is added to etcd static pod to run automated backups. 
+The objectives of the new container are:
+1.  On all etcd static pods, run a subcommand of cluster-etcd-operator to produce automated backups.
+1. On all masters it creates a new backup revision `/etc/kubernetes/static-pod-backups/backup-N`.
```

Comment:
Is the backup strategy per revision or per revision per backup interval?

Reply:
No. This revision number is the number for our backups, and has nothing to do with the actual revisions on the static pods. We take backups of the latest revision at the time of backup.